libs/metrics/test_positivity.py

```diff
@@ -36,19 +55,32 @@ def columns(self) -> Set[FieldName]:
         pass
 
     @abstractmethod
-    def calculate(self, dataset: MultiRegionDataset, diff_days: int) -> MultiRegionDataset:
+    def calculate(
+        self, dataset: MultiRegionDataset, diff_days: int, most_recent_date
+    ) -> MethodOutput:
         """Calculate a DataFrame with LOCATION_ID index and DATE columns.
 
         Args:
             df: DataFrame with rows having MultiIndex of [VARIABLE, LOCATION_ID] and columns with DATE
                 index. Must contain at least one real value for each of self.columns.
             delta_df: DataFrame with rows having MultiIndex of [VARIABLE, LOCATION_ID] and columns with DATE
                 index. Values are 7-day deltas. Must contain at least one real value for each of self.columns.
+            most_recent_date: The most recent date in dataset. Timeseries with a real value for
+                at least one of the dates within recent_days of this are considered recent.
         """
         pass
 
+    def _filter_recent(self, all: MultiRegionDataset, most_recent_date) -> MethodOutput:
```

Comment:
would prefer not to use `all` as it's a python reserved keyword.  

also, filter is a bit overloaded to me, maybe `remove_series_with_stale_data` or something similar might be more clear

Reply:
Technically I'm conflicting with `all` builtin but it isn't reserved or a keyword. Changed to `ds_all`.
Yeah, I found 'filter' confusing too. Changed to `_remove_stale_regions`.